flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/filesystem/Buckets.java

```diff
@@ -237,7 +234,7 @@ void onElement(IN value, SinkFunction.Context context) throws Exception {
 					subtaskIndex,
 					bucketId,
 					bucketPath,
-					initMaxPartCounter,
+					maxPartCounterUsed,
 					partFileWriterFactory);
 			activeBuckets.put(bucketId, bucket);
 		}
```

Comment:
can we add a comment just below that `final PartFileInfo<BucketID> info = bucket.getInProgressPartInfo();` is `null`  directly after a snapshot and for new empty buckets? This is the detail I was missing when trying to understand how this fixes the issue.

Reply:
OK.